lib/uklibid/Config.uk

```diff
@@ -1,10 +1,70 @@
-config LIBUKLIBID
+menuconfig LIBUKLIBID
 	bool "uklibid: Library identifier"
-	default n
+	default y
 	# FIXME: We actually depend on <uk/assert.h> but lib/ukdebug also
 	#        depends on us. If we specify the dependency here, it creates a
 	#        circular dependency that KConfig doesn't like. However, we
 	#        can currently assume that lib/ukdebug is always part of a
 	#        build, so we comment out the dependency here. Of course, this
 	#        is not accurate.
 	#select LIBUKDEBUG
+
+if LIBUKLIBID
+config LIBUKLIBID_INFO
+	bool "Embedded library metadata"
+	default y
+
+	if LIBUKLIBID_INFO
+	menu "Global metadata"
+		config LIBUKLIBID_INFO_UKFULLVERSION
+			bool "Full unikraft version"
+			default y
+			help
+				If enabled, the full Unikraft version including
+				Git SHA is embedded. Otherwise, only the short
+				Unikraft version is stored.
+
+		config LIBUKLIBID_INFO_UKCODENAME
+			bool "Unikraft codename"
+			default n
+			help
+				If enabled, the Unikraft release codename is
+				stored as part of the global metadata
+				information.
+	endmenu
+
+	config LIBUKLIBID_INFO_COMPILER
+		bool "Compiler (CC) information"
+		default y
+		help
+			This option enables storing of the C compiler ($(CC))
+			that was in use while a library and unikernel was
+			compiled.
+
+	config LIBUKLIBID_INFO_COMPILEDATE
+		bool "Compile date"
+		default n
+		help
+			This option stores the compile date and time for each
+			library and the unikernel image.
+
+	config LIBUKLIBID_INFO_COMPILEDBY_TEXT
+		string "Compiled by"
```

Comment:
I change it to auto-detected username (user@host) but keeping a configuration option to overwrite this with a custom string. The idea is that this could be useful for CI/CD system or for the case user@host should not be used as this information.

Regarding your second point: We can add extra free later with another PR by using the `UKLI_REC_COMMENT` record. I had the same thoughts already. We could even have multiple `UKLI_REC_COMMENT` records per library if needed.